src/main/java/attendance/controller/AttendanceController.java

```diff
@@ -0,0 +1,237 @@
+
+public class AttendanceController {
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final Attendances attendances;
+
+    public AttendanceController(InputView inputView, OutputView outputView) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        attendances = new AttendancesFactory().initialize();
+    }
+
+    public void run() {
+        boolean isQuit;
+        do {
+            LocalDateTime now = LocalDateTime.now();
+            isQuit = start(now);
+        } while (isQuit);
+    }
+
+    private boolean start(LocalDateTime now) {
+        Command command = null;
+        try {
+            command = Command.from(inputView.inputCommand(now.toLocalDate()));
+            logic(command, now);
+        } catch (DateTimeException e) {
+            outputView.printDateTimeErrorMessage();
+        } catch (RuntimeException e) {
+            outputView.printErrorMessage(e.getMessage());
+        }
+        return command != Command.QUIT;
+    }
+
+    private void logic(Command command, LocalDateTime now) {
+        if (command == Command.ATTENDANCE) {
+            doAttendance(now);
+        }
+        if (command == Command.ATTENDANCE_UPDATE) {
+            doUpdateAttendance(now);
+        }
+        if (command == Command.ATTENDANCE_TIMELINE) {
+            doAttendanceTimeline(now);
+        }
+        if (command == Command.EMERGENCY_CHECK) {
+            doEmergencyCheck(now);
+        }
+    }
+
+    private void doAttendance(LocalDateTime now) {
+        Attendance attendance = createAttendance(now);
+        attendances.add(attendance);
+        outputView.printCheckAttendance(attendance.getDateTime(), calculateAttendanceType(attendance.getDateTime()));
+    }
+
+    private Attendance createAttendance(LocalDateTime now) {
+        Crew crew = new Crew(inputExistNickname());
+        LocalDateTime attendanceDateTime = inputAttendanceDateTime(now);
+        return new Attendance(crew, attendanceDateTime);
+    }
+
+    private String inputExistNickname() {
+        String nickname = inputView.inputNickname();
+        attendances.validateExistNickname(nickname);
+        return nickname;
+    }
+
+    private LocalDateTime inputAttendanceDateTime(LocalDateTime now) {
+        String rawAttendanceTime = inputView.inputAttendanceTime();
+        LocalTime attendanceTime = toLocalTime(rawAttendanceTime);
+        return LocalDateTime.of(now.toLocalDate(), attendanceTime);
+    }
+
+    private void doUpdateAttendance(LocalDateTime now) {
+        Crew crew = new Crew(inputExistNicknameForUpdate());
+        LocalDateTime updateDateTime = inputUpdateDateTime(now);
+        Optional<Attendance> beforeAttendance = findAttendanceByCrewAndDate(crew, updateDateTime);
+        Attendance modifidedAttendance = attendances.update(new Attendance(crew, updateDateTime));
+        printModifiedAttendance(beforeAttendance.orElse(null), modifidedAttendance);
```

Comment:
이 부분은 Optional을 반환하고 메서드 파라미터로 주입할 때, orElse(null)의 결과로 내려주신 이유가 있을까요? 

아래와 같이처음부터 Optional을 벗겨내도 동일한 결과가 나오지 않을까요? 
```suggestion
        Attendance beforeAttendance = findAttendanceByCrewAndDate(crew, updateDateTime)
            .orElse(null);
        Attendance modifidedAttendance = attendances.update(new Attendance(crew, updateDateTime));
        printModifiedAttendance(beforeAttendance, modifidedAttendance);
```

Reply:
해당 부분도 리팩터링 진행하겠습니다!!